### PR TITLE
py-numpy: enable linking with armpl-gcc and acfl for BLAS and LAPACK

### PR DESCRIPTION
**Pull request by annop-w**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-numpy/package.py [hunk 1]
```diff
@@ -315,11 +315,17 @@ def write_library_dirs(f, dirs):
...
+                if spec.satisfies("+blas"):
+                    f.write("[blas]\n")
+                    f.write("libraries = {0}\n".format(lapackblas_lib_names))
+                    write_library_dirs(f, lapackblas_lib_dirs)
+                    f.write("include_dirs = {0}\n".format(lapackblas_header_dirs))
+                if spec.satisfies("+lapack"):
+                    f.write("[lapack]\n")
+                    f.write("libraries = {0}\n".format(lapackblas_lib_names))
+                    write_library_dirs(f, lapackblas_lib_dirs)
+                    f.write("include_dirs = {0}\n".format(lapackblas_header_dirs))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numpy/package.py]:
```suggestion
                if spec.satisfies("+blas"):
                    f.write("[blas]\n")
                    f.write("libraries = {0}\n".format(blas_lib_names))
                    write_library_dirs(f, blas_lib_dirs)
                    f.write("include_dirs = {0}\n".format(blas_header_dirs))
                if spec.satisfies("+lapack"):
                    f.write("[lapack]\n")
                    f.write("libraries = {0}\n".format(lapack_lib_names))
                    write_library_dirs(f, lapack_lib_dirs)
                    f.write("include_dirs = {0}\n".format(lapack_header_dirs))
```
I know there's only a single library for both blas and lapack, but I think it makes more visual sense to organize them like this.

Comment (rgommers) [reply var/spack/repos/builtin/packages/py-numpy/package.py]: > I know there's only a single library for both blas and lapack

That's not always true: Netlib BLAS/LAPACK, BLIS/libFlame, and you can even build OpenBLAS without LAPACK.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-numpy/package.py]: Yes, I'm only talking about armpl/acfl

Comment (annop-w) [reply var/spack/repos/builtin/packages/py-numpy/package.py]: @adamjstewart Thank you for the suggestion !!

**Review by adamjstewart — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by annop-w** [comment]
@alalazo Some of the pipelines failed. Does this need a re-run ? Thanks.

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Review by adamjstewart — DISMISSED** [review]

**Comment by annop-w** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by annop-w** [comment]
@alalazo Looks like the pipelines are constantly failing. How can they be fixed ?

**Comment by alalazo** [comment]
Failures in pipelines seem due to system errors. Merged the PR.

**Pull request merged**